### ISSUE TITLE
Searching only in command history

### ISSUE THREAD

**Issue by ratheesh** [gelguy/wilder.nvim#14]
Is searching only in command history supported?

**Comment by gelguy** [comment]
You can use the following pipeline:
```
call wilder#set_option('pipeline', [
      \ wilder#history(100),
      \ {ctx, xs -> filter(xs, {i, x -> match(x, ctx.input) != -1})},
      \ ])
```
You can increase the number of history candidates by increasing `100`, but note that at higher numbers there might be noticeable lag as the gathering of the history candidates is quite inefficient. `0` sets the history candidates to unlimited.

**Comment by ratheesh** [comment]
I have defined the following pipeline defined.
```vim
      call wilder#set_option('pipeline', [
                  \   wilder#branch(
                  \     [
                  \       wilder#check({_, x -> empty(x)}),
                 \       wilder#history(15),
                  \     ],
                  \     wilder#cmdline_pipeline(),
                  \     [                                                                                                                                                                       
                 \       wilder#python_fuzzy_delimiter(),
                  \       wilder#python_search({'engine': 're'}),
                  \       wilder#result_output_escape('^$,*~[]/\'),
                  \     ],
                 \     [
                 \       wilder#history(50),
                 \       {ctx, xs -> filter(xs, {i, x -> match(x, ctx.input) != -1})},
                 \     ],
                  \   ),
                  \ ])
```
The last segment is completing only history items. May I know how to bind this to a different key? example to `<BS>`

**Comment by ratheesh** [comment]
Let me rephrase my above comment.

I am looking  at an option to complete from history items for the existing command line modes - `:  / or ?`
Example: If `C-r` is pressed in command line mode, only history items should be sourced, else normal items should be sourced.
It will be immensely useful. This reduce considerable time in executing the desired command.

**Comment by gelguy** [comment]
Thanks for bringing up this use case! I was not considering this usage but this is actually quite powerful as it enables you to change the pipeline on the fly.

You can add the following to your `init.vim`/`.vimrc` to achieve the effect:
```vim
let s:wilder_show_history = 0
function! s:wilder_toggle_history() abort
  let s:wilder_show_history = 1 - s:wilder_show_history
  call timer_start(0, {-> feedkeys("\<BS>")})
  " return space which is then deleted by the <BS> to trigger CmdlineChanged
  return ' '
endfunction

cmap <expr> <C-F> wilder#in_context() ? <SID>wilder_toggle_history() : "\<C-F>"

call wilder#set_option('pipeline', [
      \   wilder#branch(
      \     [
      \       wilder#check({-> s:wilder_show_history}),
      \       wilder#history(100),
      \       {ctx, xs -> filter(xs, {i, x -> match(x, ctx.input) != -1})},
      \     ],
      \    " rest of your pipelines
      \   ),
      \ ])
```
Remember the ordering of the branches matter so the check should be done before the other pipelines.

The `s:wilder_show_history` variable can also be used in the renderer to indicate you are using history mode.

Currently, triggering the `CmdlineChanged` event is inefficient as it has to be triggered twice. I will be adding a helper method to trigger a change on demand in the future.

**Comment by ratheesh** [comment]
Thank you. This is working fine.
